### DISCUSSION TITLE
Newest version of a device uses a different battery. How to reflect this in the library?

### DISCUSSION THREAD

**Discussion by brystmar** [andrew-codechimp/HA-Battery-Notes#1899] [Q&A] [answered]
While setting up BN for the first time with my Zooz water leak sensors (ZSE42), I remembered that Zooz changed the battery for this device when updating the hardware from the 700 to 800 series z-wave chip.

This coincides with a firmware version bump: all v1.xx firmware is for the 700 series which uses the old battery (CR2032), while v2.xx firmware is for the newer 800 series with a different battery (CR2450).  This is called out explicitly in the [ZSE42 firmware changelog](https://www.support.getzooz.com/kb/article/998-zse42-water-leak-xs-sensor-change-log/).

My first thought was to add a new entry in the json library.  However the device name in HA is still "ZSE42", so per the FAQ it doesn't warrant a new entry.  I see someone added "ZSE40 700" to reflect this for the ZSE40, which apparently also changed batteries between generations.  And my remote scene controller's device name is "ZEN37 800LR" in HA, which is distinct from the "ZEN37" in the library, so I'll probably add this as a new device.  It's unfortunate that these Zooz device names aren't always consistent.

How should this be handled?

**Comment by andrew-codechimp** [comment, marked as answer]
This is a tricky one and there are a few ways to handle it.

If the manufacturer/model are identical for both revisions we can set the battery type to MANUAL to stop it incorrectly suggesting the wrong battery, you would have to add the devices manually each time and put in your own battery. I have just made this change to stop bad suggestions.

If within the HA device screen there is a hardware version and this differs I can support that with an optional hw_version field in the library, I've only had one device with that so it's not generally publicised yet but will be soon.  If there is a hw_version difference please let me know and I'll add the two entries.

With HA version 2024.8 they are introducing a model ID to allow further distinguishing between devices, but that will be down to the integration adding this field, and that's if the integration can query the device and get that data. There won't be a big bang switch to having model ID within HA but I'm currently working on support for this within the library ready for when integrations do start using this. I'm not sure what integration you are using for Zooz devices but once model ID is available perhaps contact the integration owner to see whether they can implement this.

**Reply by brystmar** [reply]
I'm using the ZWaveJS integration.  My zwave devices all have the `134-0-hardwareVersion` attribute.  My 700-series ZSE42 with the CR2032 is hw version 1.  My 800-series ZSE42 with the CR2450 is hw version 2.  I spot-checked more than a dozen of my other Zooz devices and can confirm the value for this attribute cleanly aligns with their published hardware refreshes.

Unfortunately, I don't see this `hw_version` on the device page in HA.  It does show the firmware, which maps cleanly to battery type for this particular device.

I like the enhancement for adding model ID to HA.  I'll request the ZWaveJS maintainer add the `hw_version` attribute when adopting model ID.
